scripts/generate_spm_sources_layout.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/sh
+
+# Swift Package Manager [has some strict requirements](https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#creating-c-language-targets) to source files layout.
+# We can gain SPM support with minimal project structure changes by applying an autogenerated layout based on symbolic links technics. 
+# SPM related files located under `spm` folder at project's root.
+# You should **NEVER** modify it manually nor inlude to Xcode's project.
+
+#  ### Generate SPM layout
+
+#  1. From **project's root** run:
+
+#     `bash scripts/generate_spm_sources_layout.sh`
+
+#  2. Commit Changes
+
+#  Repeate those steps each time you delete/add project's files. **Make sure** to have this CI step which will check that `generate_spm_sources_layout.sh` is not broken.
```

Comment:
Looks like we dont have this step in CI yet, maybe add it to the Github action file or the ./setup.sh too? What do you think?